### PR TITLE
fix: add checks for peripheral.mtu

### PR DESCRIPTION
**Pull request by fisherwise**
checks if it's valid when setting it.
Should fix the TypeError of it

### CODE REVIEW

**Comment by rzr** [comment]
please rebase

File: lib/noble.js [hunk 1]
```diff
@@ -569,7 +569,7 @@ Noble.prototype.onHandleNotify = function (peripheralUuid, handle, data) {
 
 Noble.prototype.onMtu = function (peripheralUuid, mtu) {
   var peripheral = this._peripherals[peripheralUuid];
-  peripheral.mtu = mtu;
+  if (peripheral && peripheral.mtu && mtu) peripheral.mtu = mtu;
```
[the remark this answers is not in the document]
Comment (rzr) [reply lib/noble.js:572]: or maybe :    var peripheral = this._peripherals[peripheralUuid] || {} ?

**Comment by rzr** [comment]
can you please try to update your branch ?

File: lib/noble.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fisherwise) [reply lib/noble.js:572]: > or maybe : var peripheral = this._peripherals[peripheralUuid] || {} ?

if ```peripheral = {}``` then ```peripheral.mtu``` will get TypeError i guess? same for ```if (peripheral)``` when accessing ```.mtu```

**Comment by fisherwise** [comment]
@rzr sry for the late reply just rebased it

File: lib/noble.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rzr) [reply lib/noble.js:589]: if peripheral = {} then peripheral.mtu is undefined but you can do affectation:

``
node
Welcome to Node.js v14.15.0.
Type ".help" for more information.
> var p = {};
undefined
> p.foo = true;
true
> console.log( p.as)
undefined


```

**Review by rzr — CHANGES_REQUESTED** [review]
could be simplified

**Review by fisherwise — COMMENTED** [review]
But this is to fix TypeError. Checked with typescript and yields error

**Pull request merged**